### PR TITLE
Add ScaleShiftLayer

### PR DESCRIPTION
**Pull request by guoshengCS**
Add trainable ScaleShiftLayer to do scaling and shifting.

### CODE REVIEW

File: paddle/gserver/tests/test_LayerGrad.cpp [hunk 1]
```diff
@@ -2007,6 +2007,21 @@ TEST(Layer, RowL2NormLayer) {
   }
 }
 
+TEST(Layer, ScaleShiftLayer) {
+  const size_t batchSize = 128;
+  const size_t size = 512;
```
Comment (lcy-seso) [inline paddle/gserver/tests/test_LayerGrad.cpp]: 这两个size 改小一些，这个layer 对size 不是非常铭感，没有必要测试这么大的layer。

File: python/paddle/trainer_config_helpers/tests/configs/test_scale_shift_layer.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from paddle.trainer_config_helpers import *
+
+settings(batch_size=1000, learning_rate=1e-5)
```
Comment (lcy-seso) [inline python/paddle/trainer_config_helpers/tests/configs/test_scale_shift_layer.py]: 这一行可以去掉。

File: python/paddle/trainer_config_helpers/layers.py [hunk 3]
```diff
@@ -6210,3 +6212,38 @@ def kmax_sequence_score_layer(input, name=None, beam_size=1):
...
+    :param bias_attr: The parameter attribute of shifting.
+    :type bias_attr: ParameterAttribute
+    :return: LayerOutput object.
+    :rtype: LayerOutput
+    """
+    Layer(
+        name=name,
+        type=LayerType.SCALE_SHIFT_LAYER,
+        inputs=Input(input.name, **param_attr.attr),
+        bias=ParamAttr.to_bias(bias_attr))
```
Comment (lcy-seso) [inline python/paddle/trainer_config_helpers/layers.py]: 有一点想确认一下，如果用户不指定 w 这个参数的初始化 std, mean 和策略，  w  这个参数会如何初始化？会不会有意外情况被设置为 0 ，后续网络可能就直接废掉了 ，默认值设置的逻辑有检查过吗 ？

File: paddle/gserver/layers/ScaleShiftLayer.cpp [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+limitations under the License. */
+
+#include "Layer.h"
+
+namespace paddle {
+
+/**
+ * A layer applies a slope and an intercept to the input element-wise for
+ * scaling and shifting. Noting that this layer is trainable which differs
+ * from the SlopeInterceptLayer.
```
Comment (lcy-seso) [inline paddle/gserver/layers/ScaleShiftLayer.cpp]: A layer applies a linear transformation to each element in each row of the input matrix. For each element, the layer first re-scale it and then adds a bias to it.

This layer is very like the SlopeInterceptLayer, except the scale and bias are trainable.

File: paddle/gserver/layers/ScaleShiftLayer.cpp [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+ *
+ * \f[
+ *    y = wx + b
+ * \f]
+ *
+ * Here, w is scale and b is offset, which are scalars and trainable.
```
Comment (lcy-seso) [inline paddle/gserver/layers/ScaleShiftLayer.cpp]: w is the scale and b is the bias. Both w and b are trainable scalars.

File: python/paddle/trainer_config_helpers/layers.py [hunk 6]
```diff
@@ -6210,3 +6212,39 @@ def kmax_sequence_score_layer(input, name=None, beam_size=1):
...
+
+
+@wrap_name_default("scale_shift")
+@wrap_param_attr_default()
+@wrap_bias_attr_default()
+def scale_shift_layer(input, name=None, param_attr=None, bias_attr=None):
+    """
+    A layer applies a slope and an intercept to the input element-wise for 
+    scaling and shifting. Noting that this layer is trainable which differs
+    from the slope_intercept_layer.
```
Comment (lcy-seso) [inline python/paddle/trainer_config_helpers/layers.py]: A layer applies a linear transformation to each element in each row of the input matrix. For each element, the layer first re-scale it and then adds a bias to it.

This layer is very like the SlopeInterceptLayer, except the scale and bias are trainable.

File: paddle/gserver/layers/ScaleShiftLayer.cpp (hunk 4 above)
Comment (guoshengCS) [inline paddle/gserver/layers/ScaleShiftLayer.cpp]: Done. Thanks.

File: paddle/gserver/layers/ScaleShiftLayer.cpp (hunk 5 above)
Comment (guoshengCS) [inline paddle/gserver/layers/ScaleShiftLayer.cpp]: Done. Thanks.

File: python/paddle/trainer_config_helpers/layers.py (hunk 6 above)
Comment (guoshengCS) [inline python/paddle/trainer_config_helpers/layers.py]: Done. Thanks.

**Pull request merged**